src/Api/Billing/Controllers/OrganizationsController.cs

```diff
@@ -0,0 +1,394 @@
+        if (!await currentContext.OrganizationOwner(orgIdGuid))
+        {
+            throw new NotFoundException();
+        }
+
+        var organization = await organizationRepository.GetByIdAsync(orgIdGuid);
+        if (organization == null)
+        {
+            throw new NotFoundException();
+        }
+
+        var taxInfo = await paymentService.GetTaxInfoAsync(organization);
+        return new TaxInfoResponseModel(taxInfo);
+    }
+
+    [HttpPut("{id}/tax")]
+    [SelfHosted(NotSelfHostedOnly = true)]
+    public async Task PutTaxInfo(string id, [FromBody] ExpandedTaxInfoUpdateRequestModel model)
+    {
+        var orgIdGuid = new Guid(id);
+        if (!await currentContext.OrganizationOwner(orgIdGuid))
+        {
+            throw new NotFoundException();
+        }
+
+        var organization = await organizationRepository.GetByIdAsync(orgIdGuid);
+        if (organization == null)
+        {
+            throw new NotFoundException();
+        }
+
+        var taxInfo = new TaxInfo
+        {
+            TaxIdNumber = model.TaxId,
+            BillingAddressLine1 = model.Line1,
+            BillingAddressLine2 = model.Line2,
+            BillingAddressCity = model.City,
+            BillingAddressState = model.State,
+            BillingAddressPostalCode = model.PostalCode,
+            BillingAddressCountry = model.Country,
+        };
+        await paymentService.SaveTaxInfoAsync(organization, taxInfo);
+    }
+
+    /// <summary>
+    /// Tries to grant owner access to the Secrets Manager for the organization
+    /// </summary>
+    /// <param name="organizationId"></param>
+    /// <param name="userId"></param>
+    private async Task TryGrantOwnerAccessToSecretsManagerAsync(Guid organizationId, Guid userId)
+    {
+        var organizationUser = await organizationUserRepository.GetByOrganizationAsync(organizationId, userId);
+
+        if (organizationUser != null)
+        {
+            organizationUser.AccessSecretsManager = true;
+            await organizationUserRepository.ReplaceAsync(organizationUser);
+        }
+    }
+
+    /// <summary>
+    /// Adjusts the organization seats for the Secrets Manager trial to match the new seat count for secrets manager
+    /// </summary>
+    /// <param name="id"></param>
+    /// <param name="organization"></param>
+    /// <param name="model"></param>
+    private async Task<Organization> AdjustOrganizationSeatsForSmTrialAsync(Guid id, Organization organization,
+        SecretsManagerSubscriptionUpdateRequestModel model)
+    {
+        if (string.IsNullOrWhiteSpace(organization.GatewayCustomerId) ||
+            string.IsNullOrWhiteSpace(organization.GatewaySubscriptionId) ||
+            model.SeatAdjustment == 0)
+        {
+            return organization;
+        }
+
+        var subscriptionInfo = await paymentService.GetSubscriptionAsync(organization);
+        if (subscriptionInfo?.CustomerDiscount?.Id != "sm-standalone")
```

Comment:
`sm-standalone` is present on `StripeConstants.cs`, could we use that here? I lean towards centralizing these types of codes within a static class so we only have to change it in one place if we ever need to.